api/server.js

```diff
@@ -185,11 +186,19 @@ app.post("/bookshelves/remove", async (req, res) => {
  * Note: All parameters are optional (Should we require at least 1 parameter?)
  * If no books match search criteria... return an empty JSON Object
  */
-app.get("/books/search/:title/:author/:genre/:offset/:resultLength", async function(req, res) {
+app.get("/books/search", async function(req, res) {
 	try {
-		
+		res.type("json");
+		let offset = req.body.offset ? req.body.offset : 0;
+		let resultLength = req.body.resultLength ? req.body.resultLength : 10;
+		let books = await getMatchingBooks(req.body);
+		res.status(SUCCESS_CODE).json({
+			remainingBooksInSearch : books.slice(offset + resultLength).length,
+			books : books.slice(offset, offset + resultLength)
```

Comment:
I would like confirmation through testing done by someone that `books.slice()` won't throw an error if `offset` is greater than the number of elements in `books`. Also that `books.slice()` won't throw an error if `offset + resultLength` is greater than the number of elements in `books`

Reply:
No errors are caused here (Pythons slice is pretty nice that way), but you can double check if you want.